pytests/test_client.py

```diff
@@ -0,0 +1,307 @@
+        )
+    )
+
+    set_up["mock_stub"].CreateGridpoolOrder.assert_called_once()
+    args, _ = set_up["mock_stub"].CreateGridpoolOrder.call_args
+    assert args[0].gridpool_id == set_up["gridpool_id"]
+    assert args[0].order.type == set_up["order_type"].to_pb()
+    assert args[0].order.quantity == set_up["quantity"].to_pb()
+    assert args[0].order.price == set_up["price"].to_pb()
+    assert args[0].order.delivery_period == set_up["delivery_period"].to_pb()
+    assert args[0].order.delivery_area == set_up["delivery_area"].to_pb()
+    assert args[0].order.execution_option == set_up["order_execution_option"].to_pb()
+
+
+def test_update_gridpool_order(
+    set_up: dict[str, Any],
+) -> None:
+    """Test the method updating a gridpool order."""
+    # Setup the expected response with valid values,
+    # especially so that the DeliveryPeriod does not raise an error
+    order_detail_response = set_up_order_detail_response(set_up)
+    mock_response = electricity_trading_pb2.UpdateGridpoolOrderResponse(
+        order_detail=order_detail_response
+    )
+    set_up["mock_stub"].UpdateGridpoolOrder.return_value = mock_response
+
+    set_up["loop"].run_until_complete(
+        set_up["client"].update_gridpool_order(
+            gridpool_id=set_up["gridpool_id"],
+            order_id=1,
+            quantity=set_up["quantity"],
+            valid_until=set_up["valid_until"],
+        )
+    )
+
+    valid_until_pb = timestamp_pb2.Timestamp()
+    valid_until_pb.FromDatetime(set_up["valid_until"])
+
+    set_up["mock_stub"].UpdateGridpoolOrder.assert_called_once()
+    args, _ = set_up["mock_stub"].UpdateGridpoolOrder.call_args
+    assert args[0].update_order_fields.quantity == set_up["quantity"].to_pb()
+    assert args[0].update_order_fields.valid_until == valid_until_pb
+    # Test that other fields e.g. price are not set
+    assert not args[0].update_order_fields.HasField(
+        "price"
+    ), "Price field should not be set."
+
+
+def test_cancel_gridpool_order(
+    set_up: dict[str, Any],
+) -> None:
+    """Test the method cancelling gridpool orders."""
+    # Setup the expected response with valid values,
+    # especially so that the DeliveryPeriod does not raise an error
+    order_detail_response = set_up_order_detail_response(set_up)
+    mock_response = electricity_trading_pb2.CancelGridpoolOrderResponse(
+        order_detail=order_detail_response
+    )
+
+    # Order to cancel
+    order_id = 1
+
+    set_up["mock_stub"].CancelGridpoolOrder.return_value = mock_response
+
+    set_up["loop"].run_until_complete(
+        set_up["client"].cancel_gridpool_order(
+            gridpool_id=set_up["gridpool_id"], order_id=order_id
+        )
+    )
+
+    set_up["mock_stub"].CancelGridpoolOrder.assert_called_once()
+    args, _ = set_up["mock_stub"].CancelGridpoolOrder.call_args
+    assert args[0].gridpool_id == set_up["gridpool_id"]
+    assert args[0].order_id == order_id
+
+
```

Comment:
Just for the record: I don't think it makes that much sense here since these functions are wrappers only and don't raise errors themselves. Given that I am mocking the underlying function testing an exception being raised would mean I need to mock the response of the graph function to return an error, and then checking that this error was indeed returned.